cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/globals_vars_test.go

```diff
@@ -0,0 +1,381 @@
+// Copyright 2022 the Kubeapps contributors.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"github.com/vmware-tanzu/kubeapps/cmd/apprepository-controller/pkg/apis/apprepository/v1alpha1"
+	corev1 "github.com/vmware-tanzu/kubeapps/cmd/kubeapps-apis/gen/core/packages/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
```

Comment:
Correct me if I'm wrong here, @absoludity , but I'd rather say we weren't so fans of globals in the tests, preferring instead each test case being defining (and customizing with a given set of test-case properties) the object.

So, what about replacing those globals with just functions creating each struct based upon a set of params (where you foresee the variability, I mean).

Reply:
Ok, I didn't know about that decision. I've used functions to generate structs at some points, but didn't want to convert tests in another set of complex code 😄

Reply:
Not so much a decision as a recommendation - global vars create shared state that is then used by different tests. Can lead to one test updating that shared state causing hard to debug failures in other tests etc. See https://dave.cheney.net/2017/06/11/go-without-package-scoped-variables for a good discussion.